experimental/dds/tree2/src/test/shared-tree-core/sharedTreeCore.spec.ts

```diff
@@ -260,40 +260,44 @@ describe("SharedTreeCore", () => {
+		// sequence number just before the most recently received changed. Thus, eviction from this point of view
+		// is "off by one"; a commit is only evicted once another commit is sequenced after it.
 		//
-		//                                 trunk: [seqNum1, (branchBaseA, branchBaseB, ...), seqNum2, ...]
+		//                                            trunk: [seqNum1, (branchBaseA, branchBaseB, ...), seqNum2, ...]
 		changeTree(tree);
-		factory.processAllMessages(); //          [1]
+		factory.processAllMessages(); //                     [1]
 		assert.equal(getTrunkLength(tree), 1);
 		const branch1 = tree.getLocalBranch().fork();
 		const branch2 = tree.getLocalBranch().fork();
 		const branch3 = branch2.fork();
 		changeTree(tree);
-		factory.processAllMessages(); //          [1 (b1, b2, b3), 2]
+		factory.processAllMessages(); //                     [x (b1, b2, b3), 2]
 		changeTree(tree);
-		factory.processAllMessages(); //          [1 (b1, b2, b3), 2, 3]
-		assert.equal(getTrunkLength(tree), 3);
-		branch1.dispose(); //                     [1 (b2, b3), 2, 3]
-		assert.equal(getTrunkLength(tree), 3);
-		branch2.dispose(); //                     [1 (b3), 2, 3]
-		assert.equal(getTrunkLength(tree), 3);
-		branch3.dispose(); //                     [x, 2, 3]
+		factory.processAllMessages(); //                     [x (b1, b2, b3), 2, 3]
 		assert.equal(getTrunkLength(tree), 2);
-		const branch4 = tree.getLocalBranch().fork(); //     [x, 2, 3 (b4)]
-		changeTree(tree);
+		branch1.dispose(); //                                [x (b2, b3), 2, 3]
+		assert.equal(getTrunkLength(tree), 2);
+		branch2.dispose(); //                                [x (b3), 2, 3]
+		assert.equal(getTrunkLength(tree), 2);
+		branch3.dispose(); //                                [x, x, 3]
+		assert.equal(getTrunkLength(tree), 1);
+		const branch4 = tree.getLocalBranch().fork(); //     [x, x, 3 (b4)]
 		changeTree(tree);
-		factory.processAllMessages(); //          [x, x, 3 (b4), 4, 5]
-		assert.equal(getTrunkLength(tree), 3);
-		const branch5 = tree.getLocalBranch().fork(); //     [x, x, 3 (b4), 4, 5 (b5)]
-		branch4.rebaseOnto(branch5); //           [x, x, 3, 4, 5 (b4, b5)]
-		branch4.dispose(); //                     [x, x, 3, 4, 5 (b5)]
-		assert.equal(getTrunkLength(tree), 3);
 		changeTree(tree);
-		factory.processAllMessages(); //          [x, x, x, 4, 5 (b5)]
+		factory.processAllMessages(); //                     [x, x, x (b4), 4, 5]
 		assert.equal(getTrunkLength(tree), 2);
-		branch5.dispose(); //                     [x, x, x, 4, 5]
+		const branch5 = tree.getLocalBranch().fork(); //     [x, x, x (b4), 4, 5 (b5)]
+		branch4.rebaseOnto(branch5); //                      [x, x, x, 4, 5 (b4, b5)]
+		branch4.dispose(); //                                [x, x, x, 4, 5 (b5)]
 		assert.equal(getTrunkLength(tree), 2);
+		changeTree(tree);
+		factory.processAllMessages(); //                     [x, x, x, x, 5 (b5)]
```

Comment:
Shouldn't this be `[x, x, x, x, 5 (b5) 6]` because of the `changeTree(tree)` on line 295?

Reply:
This will also affect line 299.

Reply:
Yes, you're right - thanks!